### ISSUE TITLE
UnicodeEncodeError: 'charmap' codec can't encode character '\U0001f4cc' in position 0: character maps to <undefined>

### ISSUE THREAD

**Issue by atc0005** [atc0005/list-emails#20]
Log message output which includes the stack trace:

```
2020-08-13 08:58:01,816 - get_email_subject_lines - Searching "Inbox" for ALL messages
--- Logging error ---
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\lib\logging\__init__.py", line 1028, in emit
    stream.write(msg + self.terminator)
  File "C:\ProgramData\Anaconda3\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\U0001f4cc' in position 97: character maps to <undefined>
Call stack:
  File "C:\Users\Public\Documents\LIB0018E\list_emails.py", line 490, in <module>
    account, mailbox, folder)
  File "C:\Users\Public\Documents\LIB0018E\list_emails.py", line 356, in get_email_subject_lines
    subject, type(subject))
Message: 'Type of %r after decoding passes: %s'
Arguments: ('📌 Staff Picks – August 2020', <class 'str'>)
--- Logging error ---
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\lib\logging\__init__.py", line 1028, in emit
    stream.write(msg + self.terminator)
  File "C:\ProgramData\Anaconda3\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\U0001f4cc' in position 88: character maps to <undefined>
Call stack:
  File "C:\Users\Public\Documents\LIB0018E\list_emails.py", line 501, in <module>
    TIMESTAMP)
  File "C:\Users\Public\Documents\LIB0018E\list_emails.py", line 395, in record_emails
    log.debug("Writing: %r of type %s", line, type(line))
Message: 'Writing: %r of type %s'
Arguments: ('📌 Staff Picks – August 2020', <class 'str'>)
Traceback (most recent call last):
  File "C:\Users\Public\Documents\LIB0018E\list_emails.py", line 501, in <module>
    TIMESTAMP)
  File "C:\Users\Public\Documents\LIB0018E\list_emails.py", line 396, in record_emails
    output_fh.write(line + '\n')
  File "C:\ProgramData\Anaconda3\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\U0001f4cc' in position 0: character maps to <undefined>
```

**Comment by atc0005** [comment]
> encodings\cp1252.py

That stands out.

Is it possible to automatically toss characters outside of the range of MySQL's UTF8 (mb3) encoding?

**Comment by atc0005** [comment]
Note:

This occurred today while one of our crew was processing a number of backlogged emails (from the past week?).

**Comment by atc0005** [comment]
https://stackoverflow.com/questions/56995919/change-python-3-7-default-encoding-from-cp1252-to-cp65001-aka-utf-8

> This is resolved when putting the following at the top of your python script. I am able to print all characters without error.
> 
> ```python
> import sys
> import io
> sys.stdout = io.TextIOWrapper(sys.stdout.detach(), encoding = 'utf-8')
> sys.stderr = io.TextIOWrapper(sys.stderr.detach(), encoding = 'utf-8')
> ```

**Comment by atc0005** [comment]
Looks like this is a duplicate report of #7. Internally this was fixed in SVN r34 and linked to deploy ticket svn-1149.

In short, I'm not sure that the above suggested changes are necessary, though I can't think of a reason that they would be harmful?

**Comment by atc0005** [comment]
Worth noting from https://www.python.org/dev/peps/pep-0540/:

> **Proposal**
> 
> Add a new UTF-8 Mode to use the UTF-8 encoding, ignore the locale encoding, and change stdin and stdout error handlers to surrogateescape.
> 
> Add the new `-X utf8` command line option and `PYTHONUTF8` environment variable. Users can explicitly activate UTF-8 Mode with the command-line option `-X utf8` or by setting the environment variable `PYTHONUTF8=1`.
> 
> This mode is disabled by default and enabled by the POSIX locale. Users can explicitly disable UTF-8 Mode with the command-line option `-X utf8=0` or by setting the environment variable `PYTHONUTF8=0`.
> 
> For standard streams, the `PYTHONIOENCODING` environment variable has priority over UTF-8 Mode.
> 
> On Windows, the `PYTHONLEGACYWINDOWSFSENCODING` environment variable (PEP 529) has the priority over UTF-8 Mode.

**Comment by atc0005** [comment]
Looks like all TODO items here were resolved with v0.1.3 release.
